.github/workflows/third-party-library-changes.yml

```diff
@@ -0,0 +1,78 @@
+name: Third Party Library Changes
+on:
+  pull_request:
+    branches:
+      - trunk
+      - "3.[89]"
+      - "[4-9].[0-9]"
+
+jobs:
+  third-party-changed-files:
+    runs-on: ubuntu-latest
+    name: Get third party library modified files
+    permissions:
+      pull-requests: read
+    outputs:
+      file_list: ${{ steps.modified_libraries.outputs.file_list }}
+      has_changed_files: ${{ steps.modified_libraries.outputs.has_changed_files }}
+    env:
+      THIRD_PARTY_LIBRARIES: (
+        "src/wp-includes/ID3"
+        "src/wp-includes/IXR"
+        "src/wp-includes/PHPMailer"
+        "src/wp-includes/Requests"
+        "src/wp-includes/SimplePie"
+        "src/wp-includes/sodium_compat"
+        "src/wp-includes/Text"
+        )
+    steps:
+      - name: Get changed files
+        id: changed_files
```

Comment:
I don't hate relying on this 3rd-party action. But I do wonder if there's some way to use `grep` and `git diff` to quickly match for any matched paths.